### ISSUE TITLE
`cmake` include file search out of sync with a `stdenv` update some time ago

### ISSUE THREAD

**Issue by 7c6f434c** [NixOS/nixpkgs#18042]
Currently `cmake` has a setup-hook that tries to add include search directories listed in `NIX_CFLAGS_COMPILE` using `-I` to `CMAKE_INCLUDE_PATH` so that `cmake` can find the headers when running `FIND_PACKAGE` directive.

Unfortunately, that hook is out of sync with stdenv that currently uses `-isystem /path/to/include/dir/` format.

I have added approximately this code to one package:

``` sh
preConfigure = ''
    include=
    for i in $NIX_CFLAGS_COMPILE; do
      if test -n "$include" && test -d "$i"; then
        export CMAKE_INCLUDE_PATH="$CMAKE_INCLUDE_PATH''${CMAKE_INCLUDE_PATH:+:}$i"
      fi;
      if test "x$i" = "x-isystem"; then
        include=1
      else
        include=
      fi
    done
'';
```

I think something like that is needed in the `cmake` setup hook, but the question is «when» and «in which branch» (given the talk of `release-16.09` branch deadline approaching).

**Comment by vcunat** [comment]
If you know a solution, push it to staging. Afterwards we can discuss whether it's worth picking to 16.09 which has been branched off in the meantime.

**Comment by 7c6f434c** [comment]
My only test case depends on Qt which also depends on CMake…

I hope the overnight build will succeed.

**Comment by 7c6f434c** [comment]
Pushed to `staging`

**Comment by vcunat** [comment]
Ref: dabf09aa9, in master now. I think it's solved, except for the 16.09 question.

**Comment by luke-clifton** [comment]
This issue can be closed now I think.
